watchtower/wtdb/client_db.go

```diff
@@ -2584,35 +2613,42 @@ func (c *ClientDB) filterClientSessionAcks(sessionBkt,
 
 // filterClientSessionCommits retrieves all committed updates for the session
 // identified by the serialized session id and passes them to the given
-// PerCommittedUpdateCB callback.
+// PerCommittedUpdateCB callback. If the countUpdates argument is true, then
+// the number of committed updates will be returned.
 func filterClientSessionCommits(sessionBkt kvdb.RBucket, s *ClientSession,
-	cb PerCommittedUpdateCB) error {
+	cb PerCommittedUpdateCB, countUpdates bool) (uint16, error) {
 
-	if cb == nil {
-		return nil
+	if cb == nil && !countUpdates {
```

Comment:
If that check was removed, we wouldn't need the boolean option and make the config options easier to deal with, right? Does it save a lot to exit early here?

Reply:
you're totally right! Good catch :) removed it 🎉